### PR TITLE
Update kPuct, replace with progressive widening formula from AZ paper.  Fixes #706

### LINKED ISSUE #706: implement progressive widening per paper
Now that we're interested in high-playout games, we should adjust the U term to grow with N.

### PR DESCRIPTION
**Pull request by amj**
Ok, got the progressive widening formula implemented as described in the paper.  My questions are: 1. am i doing it right? ;) and 2. did i make it slow?

### CODE REVIEW

File: cc/mcts_node.cc [hunk 1]
```diff
@@ -413,7 +415,8 @@ void MctsNode::PruneChildren(Coord c) {
 
 std::array<float, kNumMoves> MctsNode::CalculateChildActionScore() const {
   float to_play = position.to_play() == Color::kBlack ? 1 : -1;
-  float U_scale = kPuct * std::sqrt(std::max<float>(1, N() - 1));
+  float uct = 2 * std::log((1.0f + N() + kUct_base) / kUct_base) + kUct_init;
```
Comment (sethtroisi) [inline cc/mcts_node.cc]: consider exact to a function or macro

**Comment by sethtroisi** [comment]
consider adding a TODO in the python code (or removing all the python code)

File: cc/constants.h [hunk 2]
```diff
@@ -34,7 +34,11 @@ constexpr float kDefaultKomi = 7.5;
 
 constexpr float kDirichletAlpha = 0.03f * 361 / (kN * kN);
 
-static constexpr float kPuct = 2.50;
+// static constexpr float kPuct = 2.50;
+static constexpr float kUct_base = 19652;
+
+// We keep our -1 to 1 scale in the tree.
+static constexpr float kUct_init = 2.50;
```
Comment (sethtroisi) [inline cc/constants.h]: why not leave this unchanged and multiply the whole result by 2?

Comment (amj) [reply cc/constants.h]: done.

**Comment by amj** [comment]
Added the python changes.

File: cc/mcts_node.cc [hunk 3]
```diff
@@ -413,7 +415,8 @@ void MctsNode::PruneChildren(Coord c) {
 
 std::array<float, kNumMoves> MctsNode::CalculateChildActionScore() const {
   float to_play = position.to_play() == Color::kBlack ? 1 : -1;
-  float U_scale = kPuct * std::sqrt(std::max<float>(1, N() - 1));
+  float uct = 2 * (std::log((1.0f + N() + kUct_base) / kUct_base) + kUct_init);
+  float U_scale = uct * std::sqrt(std::max<float>(1, N() - 1));
```
Comment (tommadams) [inline cc/mcts_node.cc]: Consider making adding an inline `U_scale` method to `MctsNode`.

Comment (amj) [reply cc/mcts_node.cc]: Done.

File: cc/mcts_node.cc (hunk 1 above)
Comment (amj) [reply cc/mcts_node.cc]: Done -- i presevered the N/N-1 behavior for children & renamed that part to 'U_common' instead of scale.

**Pull request merged**